# Diabetes Risk Data Exploration (Introduction)

In this project we are working with a diabetes dataset to understand what affects a person’s risk of getting diabetes. The goal is to explore the data, see patterns, and understand the different features before moving to modelling.

# Importing Libraries

In [14]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
sns.set(style="whitegrid")
#sns.set(style="whitegrid") sets a clean background style for the graphs making them easier to read. 

# Loading the Dataset

In this step, we will load the dataset so we can start exploring it. This allows us to see what the data looks like and understand the different columns.

In [19]:
df = pd.read_csv("../data/Diabetes_and_LifeStyle_Dataset_.csv")
df.head(10)

,Age,gender,ethnicity,education_level,income_level,employment_status,smoking_status,alcohol_consumption_per_week,physical_activity_minutes_per_week,diet_score,sleep_hours_per_day,screen_time_hours_per_day,family_history_diabetes,hypertension_history,cardiovascular_history,bmi,waist_to_hip_ratio,systolic_bp,diastolic_bp,heart_rate,cholesterol_total,hdl_cholesterol,ldl_cholesterol,triglycerides,glucose_fasting,glucose_postprandial,insulin_level,hba1c,diabetes_risk_score,diabetes_stage,diagnosed_diabetes
0,58,Male,Asian,Highschool,Lower-Middle,Employed,Never,0,215,5.7,7.9,7.9,0,0,0,30.5,0.89,134,78,68,239,41,160,145,136,236,6.36,8.18,29.6,Type 2,1
1,52,Female,White,Highschool,Middle,Employed,Former,1,143,6.7,6.5,8.7,0,0,0,23.1,0.80,129,76,67,116,55,50,30,93,150,2.00,5.63,23.0,No Diabetes,0
2,60,Male,Hispanic,Highschool,Middle,Unemployed,Never,1,57,6.4,10.0,8.1,1,0,0,22.2,0.81,115,73,74,213,66,99,36,118,195,5.07,7.51,44.7,Type 2,1
3,74,Female,Black,Highschool,Low,Retired,Never,0,49,3.4,6.6,5.2,0,0,0,26.8,0.88,120,93,68,171,50,79,140,139,253,5.28,9.03,38.2,Type 2,1
4,46,Male,White,Graduate,Middle,Retired,Never,1,109,7.2,7.4,5.0,0,0,0,21.2,0.78,92,67,67,210,52,125,160,137,184,12.74,7.20,23.5,Type 2,1
5,46,Female,White,Highschool,Upper-Middle,Employed,Never,2,124,9.0,6.2,5.4,0,0,0,26.1,0.85,95,81,57,218,61,119,179,100,133,8.77,6.03,23.5,Pre-Diabetes,0
6,75,Female,White,Graduate,Upper-Middle,Retired,Never,0,53,9.2,7.8,8.0,0,1,0,25.1,0.88,129,77,81,238,46,161,155,101,100,10.14,5.24,36.1,Pre-Diabetes,0
7,62,Male,White,Postgraduate,Middle,Unemployed,Current,1,75,4.1,9.0,12.9,0,1,1,23.9,0.86,128,83,76,241,49,159,120,110,189,8.96,7.04,34.2,Type 2,1
8,37,Male,Black,Highschool,Lower-Middle,Employed,Current,1,114,6.7,8.5,8.5,0,0,1,24.7,0.84,103,71,72,187,33,132,98,116,172,5.70,6.90,26.7,Type 2,1
9,59,Female,White,Graduate,Middle,Employed,Current,3,86,8.2,5.3,7.4,0,0,0,26.7,0.81,124,81,70,188,52,103,104,76,109,4.49,4.99,30.0,No Diabetes,0


# Observation

The dataset contains 31 columns which include both lifestyle and medical information.

# Data Inspection

Here we will check the structure of the dataset, including the number of rows, column types, and summary statistics.

In [16]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 97297 entries, 0 to 97296
Data columns (total 31 columns):
 #   Column                              Non-Null Count  Dtype  
---  ------                              --------------  -----  
 0   Age                                 97297 non-null  int64  
 1   gender                              97297 non-null  str    
 2   ethnicity                           97297 non-null  str    
 3   education_level                     97297 non-null  str    
 4   income_level                        97297 non-null  str    
 5   employment_status                   97297 non-null  str    
 6   smoking_status                      97297 non-null  str    
 7   alcohol_consumption_per_week        97297 non-null  int64  
 8   physical_activity_minutes_per_week  97297 non-null  int64  
 9   diet_score                          97297 non-null  float64
 10  sleep_hours_per_day                 97297 non-null  float64
 11  screen_time_hours_per_day           97297 non-null  

# Understanding the Dataset Structure (df.info())

In this step, we use df.info() to understand the structure of the dataset.

This helps us see:
- the number of rows and columns
- the data types of each column (e.g. integer, float, string)
- whether there are any missing values
This is important because it helps us understand the quality of the data before doing further analysis.

# Observation (df.info())

The dataset has 97,297 rows and 31 columns. All columns have the same number of values, which means there are no missing values.

The dataset includes a mix of data types:
- numerical data (e.g. age, BMI, glucose levels)
- categorical data (e.g. gender, smoking status)

This shows that the dataset is complete and ready for analysis.

In [20]:
df.describe()

#This cell code helps us understand:
#- the average (mean) values
#- how spread out the data is (standard deviation)
#- the minimum and maximum values
#- the distribution of the data (quartiles)

,Age,alcohol_consumption_per_week,physical_activity_minutes_per_week,diet_score,sleep_hours_per_day,screen_time_hours_per_day,family_history_diabetes,hypertension_history,cardiovascular_history,bmi,waist_to_hip_ratio,systolic_bp,diastolic_bp,heart_rate,cholesterol_total,hdl_cholesterol,ldl_cholesterol,triglycerides,glucose_fasting,glucose_postprandial,insulin_level,hba1c,diabetes_risk_score,diagnosed_diabetes
count,97297.000000,97297.000000,97297.000000,97297.000000,97297.000000,97297.000000,97297.000000,97297.000000,97297.000000,97297.000000,97297.000000,97297.000000,97297.000000,97297.000000,97297.000000,97297.000000,97297.000000,97297.000000,97297.000000,97297.000000,97297.000000,97297.000000,97297.000000,97297.000000
mean,50.192699,2.004543,118.942886,5.995214,6.997708,5.997031,0.219657,0.250912,0.079067,25.615093,0.856083,115.800127,75.233964,69.628807,185.984234,54.043537,103.000308,121.466407,111.121792,160.014348,9.063591,6.520338,30.222027,0.600039
std,15.492557,1.418060,84.441920,1.780096,1.093726,2.469805,0.414017,0.433540,0.269845,3.588387,0.046848,14.287977,8.206551,8.373731,32.013395,10.275047,33.390145,43.393422,13.590382,30.938915,4.956342,0.813710,9.065063,0.489892
min,19.000000,0.000000,0.000000,0.000000,3.000000,0.500000,0.000000,0.000000,0.000000,15.000000,0.670000,90.000000,50.000000,40.000000,100.000000,20.000000,50.000000,30.000000,60.000000,70.000000,2.000000,4.000000,2.700000,0.000000
25%,38.000000,1.000000,57.000000,4.800000,6.300000,4.300000,0.000000,0.000000,0.000000,23.200000,0.820000,106.000000,70.000000,64.000000,164.000000,47.000000,78.000000,91.000000,102.000000,139.000000,5.090000,5.970000,23.800000,0.000000
50%,51.000000,2.000000,100.000000,6.000000,7.000000,6.000000,0.000000,0.000000,0.000000,25.600000,0.860000,116.000000,75.000000,70.000000,186.000000,54.000000,102.000000,121.000000,111.000000,160.000000,8.790000,6.520000,29.000000,1.000000
75%,61.000000,3.000000,160.000000,7.200000,7.700000,7.700000,0.000000,1.000000,0.000000,28.000000,0.890000,125.000000,81.000000,75.000000,208.000000,61.000000,126.000000,151.000000,120.000000,181.000000,12.450000,7.070000,35.600000,1.000000
max,90.000000,10.000000,833.000000,10.000000,10.000000,16.800000,1.000000,1.000000,1.000000,39.200000,1.060000,179.000000,110.000000,105.000000,318.000000,98.000000,263.000000,344.000000,172.000000,287.000000,32.220000,9.800000,67.200000,1.000000


In [18]:
pd.set_option('display.max_columns', None)
#It tells pandas to show all the columns and not hide anything

# Observation (df.describe())

From the summary statistics, we can see that the average age is around 50 years, which means most patients are middle-aged. The average BMI is around 25.6, which is close to the overweight range and may increase diabetes risk. Glucose levels are also slightly high on average (fasting ≈ 111), which suggests that many patients may already be at risk. Some variables such as physical activity have a very wide range (0 to over 800 minutes) which indicates possible outliers or extreme values.